site-builder/src/main.rs

```diff
@@ -206,12 +206,56 @@ mod default {
     }
 }
 
+/// Returns the path if it is `Some` or any of the default paths if they exist (attempt in order).
+pub fn path_or_defaults_if_exist() -> Option<PathBuf> {
+    // Construct the default paths. Warning! The order is important.
+    let sites_config_name = "sites-config.yaml";
+    let mut defaults: Vec<PathBuf> = vec![];
+    if let Ok(xdg_config_home) = std::env::var("XDG_CONFIG_HOME") {
+        defaults.push(
+            // $XDG_CONFIG_HOME/walrus/sites-config.yaml
+            PathBuf::from(xdg_config_home)
+                .join("walrus")
+                .join("sites-config.yaml"),
+        );
+    }
+    // Define the home directory. If it doesn't exist, use `/`.
+    let home_dir = match home::home_dir() {
+        Some(dir) => dir,
+        None => PathBuf::from("/"),
+    };
```

Comment:
Question: why use root is home does not exist?

Reply:
i'd probably skip it if it is not there 🤔

Reply:
If the home directory does not exist, I've seen systems that `$HOME` would then point to `/`. 

Hm, how would you skip it? 💭 
What would be the base of the other paths if no home dir is specified?

Reply:
>Hm, how would you skip it? 💭

simply like this: https://github.com/MystenLabs/walrus/blob/c567eef5179776756ea7b0d4c0c930830695c41a/crates/walrus-service/src/common/utils.rs#L534-L536

if there is a home, look for the `~/.config...` path, otherwise don't add it to the list of path to search from.